### PR TITLE
Add ability to control alpha pre-multiplication

### PR DESCRIPTION
**Pull request by romainguy**
The new `ImageRequest` API maps to the equivalent API in `BitmapFactory.Options` and `ImageDecoder`.
This PR addresses issue #568.

### CODE REVIEW

File: coil-base/src/main/java/coil/request/DefaultRequestOptions.kt [hunk 1]
```diff
@@ -26,6 +26,7 @@ class DefaultRequestOptions(
     val bitmapConfig: Bitmap.Config = Utils.DEFAULT_BITMAP_CONFIG,
     val allowHardware: Boolean = true,
     val allowRgb565: Boolean = false,
+    val premultipliedAlpha: Boolean = true,
```
Comment (colinrtwhite) [inline coil-base/src/main/java/coil/request/DefaultRequestOptions.kt:29]: I would only add this attribute to `ImageRequest` and `Options` unless you feel there's a good use case for setting the `ImageLoader`'s default. For context, we don't support setting the default `ColorSpace` for an `ImageLoader` either.

File: coil-base/src/main/java/coil/request/ImageRequest.kt [hunk 2]
```diff
@@ -114,9 +114,12 @@ class ImageRequest private constructor(
     /** @see Builder.allowHardware */
     val allowHardware: Boolean,
 
-    /** @see Builder.allowHardware */
+    /** @see Builder.allowRgb565 */
```
Comment (colinrtwhite) [inline coil-base/src/main/java/coil/request/ImageRequest.kt:117]: Nice catch!

File: coil-base/src/main/java/coil/decode/Options.kt [hunk 3]
```diff
@@ -50,13 +54,14 @@ class Options(
...
         allowRgb565: Boolean = this.allowRgb565,
+        premultipliedAlpha: Boolean = this.premultipliedAlpha,
         headers: Headers = this.headers,
         parameters: Parameters = this.parameters,
         memoryCachePolicy: CachePolicy = this.memoryCachePolicy,
         diskCachePolicy: CachePolicy = this.diskCachePolicy,
         networkCachePolicy: CachePolicy = this.networkCachePolicy
-    ) = Options(context, config, colorSpace, scale, allowInexactSize, allowRgb565, headers, parameters,
-        memoryCachePolicy, diskCachePolicy, networkCachePolicy)
+    ) = Options(context, config, colorSpace, scale, allowInexactSize, allowRgb565, premultipliedAlpha,
```
Comment (colinrtwhite) [inline coil-base/src/main/java/coil/decode/Options.kt:63]: We'll need to add `@Deprecated(HIDDEN)` versions of the constructor + copy method for binary compatibility.

**Comment by colinrtwhite** [comment]
Looks good - thanks for adding this! Left a few comments.

File: coil-base/src/main/java/coil/request/DefaultRequestOptions.kt (hunk 1 above)
Comment (romainguy) [reply coil-base/src/main/java/coil/request/DefaultRequestOptions.kt:29]: 👍

File: coil-base/src/main/java/coil/decode/Options.kt (hunk 3 above)
Comment (romainguy) [reply coil-base/src/main/java/coil/decode/Options.kt:63]: Will do!

**Review by colinrtwhite — APPROVED** [review]
Great, thanks!

**Pull request merged**